controllers/dbaas.redhat.com/dbaasprovider_reconciler.go

```diff
@@ -0,0 +1,286 @@
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// bridgeProviderCR CR for crunchy bridge registration
+func bridgeProviderCR(clusterRoleList *rbac.ClusterRoleList) *dbaasoperator.DBaaSProvider {
+	instance := &dbaasoperator.DBaaSProvider{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: NAME,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+
+					APIVersion:         "rbac.authorization.k8s.io/v1",
+					Kind:               "ClusterRole",
+					UID:                clusterRoleList.Items[0].GetUID(), // doesn't really matter which 'item' we use
+					Name:               clusterRoleList.Items[0].Name,
+					Controller:         pointer.BoolPtr(true),
+					BlockOwnerDeletion: pointer.BoolPtr(false),
+				},
+			},
+			Labels: labels,
+		},
+
+		Spec: dbaasoperator.DBaaSProviderSpec{
+			Provider: dbaasoperator.DatabaseProvider{
+				Name:               PROVIDER,
+				DisplayName:        DISPLAYNAME,
+				DisplayDescription: DISPLAYDESCRIPTION,
+				Icon: dbaasoperator.ProviderIcon{
+					Data:      ICONDATA,
+					MediaType: MEDIATYPE,
+				},
+			},
+			InventoryKind:  INVENTORYDATAVALUE,
+			ConnectionKind: CONNECTIONDATAVALUE,
+			CredentialFields: []dbaasoperator.CredentialField{
+				{
+					Key:         KEYFIELDNAME,
+					DisplayName: KEYFIELDDISPLAYNAME,
+					Type:        "string",
+					Required:    true,
+				},
+				{
+					Key:         SECRETFIELDNAME,
+					DisplayName: SECRETFIELDDISPLAYNAME,
+					Type:        "maskedstring",
+					Required:    true,
+				},
+			},
+		},
+	}
+	return instance
+}
+
+// CheckCrdInstalled checks whether dbaas provider CRD, has been created yet
+func (r *DBaaSProviderReconciler) checkCrdInstalled(groupVersion, kind string) (bool, error) {
+	resources, err := r.Clientset.Discovery().ServerResourcesForGroupVersion(groupVersion)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	for _, r := range resources.APIResources {
+		if r.Kind == kind {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DBaaSProviderReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	log := r.Log.WithValues("during", "DBaaSProviderReconciler setup")
+
+	// envVar set in controller-manager's Deployment YAML
+	if operatorInstallNamespace, found := os.LookupEnv("INSTALL_NAMESPACE"); !found {
+		err := fmt.Errorf("INSTALL_NAMESPACE must be set")
+		log.Error(err, "error fetching envVar")
+		return err
+	} else {
+		r.operatorInstallNamespace = operatorInstallNamespace
+	}
+
+	// envVar set for all operators
+	if operatorNameEnvVar, found := os.LookupEnv("OPERATOR_CONDITION_NAME"); !found {
+		err := fmt.Errorf("OPERATOR_CONDITION_NAME must be set")
+		log.Error(err, "error fetching envVar")
+		return err
+	} else {
+		r.operatorNameVersion = operatorNameEnvVar
+	}
+
+	customRateLimiter := workqueue.NewItemExponentialFailureRateLimiter(30*time.Second, 30*time.Minute)
```

Comment:
Once @wilybrace has time to review, let's also touch base here and make sure they're ok with the intervals and max we've used.

Reply:
Pretty fine with the intervals as defined and as exponential backoff. As I mentioned previously, longer term we would likely drive the updates from in-platform data structures spawned by the "bridge cluster controller" side, so network hits to the API are consolidated.